From 6ac4ac5428761b96751835e8cfd7f01cf1860f4e Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 16 Jan 2013 18:53:05 -0500
Subject: [PATCH] Move Spigot initialize code to its own class and call it on
 /reload. Fixes /tps and /restart breaking on reload. Also
 fixes /reload not reloading timings setting Also enforce
 that chunk-gc stays enabled, so that people who don't
 understand it don't turn it off.

---
 .../java/org/bukkit/craftbukkit/CraftServer.java   |   23 ++-----------
 src/main/java/org/bukkit/craftbukkit/Spigot.java   |   34 ++++++++++++++++++++
 2 files changed, 37 insertions(+), 20 deletions(-)
 create mode 100644 src/main/java/org/bukkit/craftbukkit/Spigot.java

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 047378a..b8e9085 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -223,26 +223,7 @@ public final class CraftServer implements Server {
         updater.check(serverVersion);
 
         // Spigot start
-        commandMap.register("bukkit", new org.bukkit.craftbukkit.command.RestartCommand("restart"));
-        commandMap.register("bukkit", new org.bukkit.craftbukkit.command.TicksPerSecondCommand("tps"));
-
-        int timeout = configuration.getInt("settings.timeout-time", 300);
-        if (timeout == 180) {
-            timeout = 300;
-            getLogger().info("Migrating to new timeout time of 300");
-            configuration.set("settings.timeout-time", timeout);
-            saveConfig();
-        }
-        org.bukkit.craftbukkit.util.WatchdogThread.startThread(timeout, configuration.getBoolean("settings.restart-on-crash", false));
-
-        whitelistMessage = configuration.getString("settings.whitelist-message", whitelistMessage);
-        stopMessage = configuration.getString("settings.stop-message", stopMessage);
-        logCommands = configuration.getBoolean("settings.log-commands", true);
-        ipFilter = configuration.getBoolean("settings.filter-unsafe-ips", false);
-        commandComplete = configuration.getBoolean("settings.command-complete", true);
-        spamGuardExclusions = configuration.getStringList("settings.spam-exclusions");
-
-        org.bukkit.craftbukkit.util.LightningSimulator.configure(configuration);
+        Spigot.initialize(this, commandMap, configuration);
 
         try {
             configuration.save(getConfigFile());
@@ -566,6 +547,7 @@ public final class CraftServer implements Server {
 
         ((DedicatedServer) console).propertyManager = config;
 
+        ((SimplePluginManager) pluginManager).useTimings(configuration.getBoolean("settings.plugin-profiling")); // Spigot
         boolean animals = config.getBoolean("spawn-animals", console.getSpawnAnimals());
         boolean monsters = config.getBoolean("spawn-monsters", console.worlds.get(0).difficulty > 0);
         int difficulty = config.getInt("difficulty", console.worlds.get(0).difficulty);
@@ -628,6 +610,7 @@ public final class CraftServer implements Server {
                 "This plugin is not properly shutting down its async tasks when it is being reloaded.  This may cause conflicts with the newly loaded version of the plugin"
             ));
         }
+        Spigot.initialize(this, commandMap, configuration); // Spigot
         loadPlugins();
         enablePlugins(PluginLoadOrder.STARTUP);
         enablePlugins(PluginLoadOrder.POSTWORLD);
diff --git a/src/main/java/org/bukkit/craftbukkit/Spigot.java b/src/main/java/org/bukkit/craftbukkit/Spigot.java
new file mode 100644
index 0000000..a535b69
--- /dev/null
+++ b/src/main/java/org/bukkit/craftbukkit/Spigot.java
@@ -0,0 +1,34 @@
+package org.bukkit.craftbukkit;
+
+import org.bukkit.command.SimpleCommandMap;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+public class Spigot {
+    public static void initialize(CraftServer server, SimpleCommandMap commandMap, YamlConfiguration configuration) {
+        commandMap.register("bukkit", new org.bukkit.craftbukkit.command.RestartCommand("restart"));
+        commandMap.register("bukkit", new org.bukkit.craftbukkit.command.TicksPerSecondCommand("tps"));
+
+        int timeout = configuration.getInt("settings.timeout-time", 300);
+        if (timeout == 180) {
+            timeout = 300;
+            server.getLogger().info("Migrating to new timeout time of 300");
+            configuration.set("settings.timeout-time", timeout);
+            server.saveConfig();
+        }
+        org.bukkit.craftbukkit.util.WatchdogThread.startThread(timeout, configuration.getBoolean("settings.restart-on-crash", false));
+
+        server.whitelistMessage = configuration.getString("settings.whitelist-message", server.whitelistMessage);
+        server.stopMessage = configuration.getString("settings.stop-message", server.stopMessage);
+        server.logCommands = configuration.getBoolean("settings.log-commands", true);
+        server.ipFilter = configuration.getBoolean("settings.filter-unsafe-ips", false);
+        server.commandComplete = configuration.getBoolean("settings.command-complete", true);
+        server.spamGuardExclusions = configuration.getStringList("settings.spam-exclusions");
+
+        if (server.chunkGCPeriod == 0) {
+            server.getLogger().severe("[Spigot] You should not disable chunk-gc. Resetting period-in-ticks to 600 ticks.");
+            server.chunkGCPeriod = 600;
+        }
+
+        org.bukkit.craftbukkit.util.LightningSimulator.configure(configuration);
+    }
+}
-- 
1.7.10.4

